core/iwasm/aot/aot_loader.c

```diff
@@ -1640,23 +1805,48 @@ load_types(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
             goto fail;
         }
 
-        if (parent_type_idx != (uint32)-1) { /* has parent */
-            AOTType *parent_type = module->types[parent_type_idx];
-            if (!wasm_type_is_subtype_of(module->types[i], parent_type,
-                                         module->types, i)) {
-                set_error_buf(error_buf, error_buf_size,
-                              "sub type does not match super type");
-                goto fail;
+        if ((rec_count == 0) || (rec_idx == rec_count - 1)) {
+            if (rec_count == 0) {
+                bh_assert(rec_idx == 0);
             }
 
-            module->types[i]->parent_type = parent_type;
-            module->types[i]->root_type = parent_type->root_type;
-            module->types[i]->inherit_depth = parent_type->inherit_depth + 1;
+            for (j = i - rec_idx; j <= i; j++) {
+                AOTType *cur_type = module->types[j];
+                parent_type_idx = cur_type->parent_type_idx;
+                if (parent_type_idx != (uint32)-1) { /* has parent */
+                    AOTType *parent_type = module->types[parent_type_idx];
+
+                    module->types[j]->parent_type = parent_type;
+                    module->types[j]->root_type = parent_type->root_type;
+                    module->types[j]->inherit_depth =
+                        parent_type->inherit_depth + 1;
+                }
+                else {
+                    module->types[j]->parent_type = NULL;
+                    module->types[j]->root_type = module->types[j];
+                    module->types[j]->inherit_depth = 0;
+                }
+            }
         }
-        else {
-            module->types[i]->parent_type = NULL;
-            module->types[i]->root_type = module->types[i];
-            module->types[i]->inherit_depth = 0;
+
+        if ((rec_count == 0) || (rec_idx == rec_count - 1)) {
+            if (rec_count == 0) {
+                bh_assert(rec_idx == 0);
+            }
```

Comment:
Duplicated code with L1808 to L1811, how about removing L1831 to L1835?

Reply:
Done